src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -851,62 +1233,247 @@ public override IOperation VisitConditionalAccessInstance(IConditionalAccessInst
+            {
+                RegionBuilder filterAndHandlerRegion = null;
 
-            AppendNewBlock(@break);
+                IOperation exceptionDeclarationOrExpression = catchClause.ExceptionDeclarationOrExpression;
+                IOperation filter = catchClause.Filter;
+                var catchBlock = new BasicBlock(BasicBlockKind.Block);
+
+                if (filter != null)
+                {
+                    filterAndHandlerRegion = new RegionBuilder(ControlFlowGraph.RegionKind.FilterAndHandler, tryAndCatchRegion, catchClause.ExceptionType, catchClause.Locals);
+                    var filterRegion = new RegionBuilder(ControlFlowGraph.RegionKind.Filter, filterAndHandlerRegion, catchClause.ExceptionType);
+
+                    EnterRegion(filterAndHandlerRegion);
+                    EnterRegion(filterRegion);
+                    AddExceptionStore(catchClause.ExceptionType, exceptionDeclarationOrExpression);
+
+                    VisitConditionalBranch(filter, ref catchBlock, sense: true);
+                    LeaveRegion(filterRegion);
+
+                    BasicBlock lastBlockInFilter = filterRegion.Blocks.Last();
+                    lastBlockInFilter.InternalNext.Flags = BasicBlock.BranchFlags.StructuredExceptionHandling;
+                    Debug.Assert(lastBlockInFilter.InternalNext.Destination == null);
+                    Debug.Assert(filterRegion.Blocks.First().Predecessors.IsEmpty);
+                }
+
+                var handlerRegion = new RegionBuilder(ControlFlowGraph.RegionKind.Handler, filterAndHandlerRegion ?? tryAndCatchRegion, catchClause.ExceptionType,
+                                                      filterAndHandlerRegion == null ? catchClause.Locals : default);
+                EnterRegion(handlerRegion);
+
+                AppendNewBlock(catchBlock, linkToPrevious: false);
+
+                if (filter == null)
+                {
+                    AddExceptionStore(catchClause.ExceptionType, exceptionDeclarationOrExpression);
+                }
+
+                VisitStatement(catchClause.Handler);
+                LinkBlocks(CurrentBasicBlock, afterTryCatchFinally);
+
+                LeaveRegion(handlerRegion);
+
+                if (filterAndHandlerRegion != null)
+                {
+                    LeaveRegion(filterAndHandlerRegion);
+                    Debug.Assert(filterAndHandlerRegion.Regions[0].Blocks.Last().InternalNext.Destination == null);
+                    Debug.Assert(handlerRegion.Blocks.First().Predecessors.All(p => filterAndHandlerRegion.Regions[0].Blocks.Contains(p)));
+                }
+                else
+                {
+                    Debug.Assert(handlerRegion.Blocks.First().Predecessors.IsEmpty);
+                }
+            }
+
+            if (tryAndCatchRegion != null)
+            {
+                LeaveRegion(tryAndCatchRegion);
+            }
+
+            if (operation.Finally != null)
```

Comment:
Everywhere else, you're comparing the region created to represent the finally to null, not comparing the finally again.

Reply:
> Everywhere else, you're comparing the region created to represent the finally to null, not comparing the finally again.

Not sure what places do you refer to?

---
In reply to: [167702031](https://github.com/dotnet/roslyn/pull/24770#discussion_r167702031) [](ancestors = 167702031)

Reply:
BTW, this block is to deal with finally that is why it check if we have one. It might do something about regions as well.

---
In reply to: [167704355](https://github.com/dotnet/roslyn/pull/24770#discussion_r167704355) [](ancestors = 167704355,167702031)